core-processor/src/ext.rs

```diff
@@ -873,12 +873,12 @@ impl Ext {
     /// Inner alloc realization.
     pub fn alloc_inner<G: GrowHandler>(
         &mut self,
-        pages: WasmPage,
+        pages_num: u32,
         mem: &mut impl Memory,
     ) -> Result<WasmPage, ProcessorAllocError> {
         self.context
             .allocations_context
-            .alloc::<G>(pages, mem, |pages| {
+            .alloc::<G>(pages_num, mem, |pages| {
```

Comment:
better to convert here `pages_num` to `page: WasmPage` and then use it as argument in `allocation_conext.alloc`